frontend/src/metabase/admin/performance/hooks/useSaveStrategy.tsx

```diff
@@ -52,17 +62,29 @@ export const useSaveStrategy = (
         const validatedStrategy =
           strategies[values.type].validateWith.validateSync(newStrategy);
 
-        const newConfig = {
+        const newConfig: CacheConfig = {
           ...baseConfig,
           strategy: validatedStrategy,
         };
 
         const translatedConfig = translateConfigToAPI(newConfig);
         await CacheConfigApi.update(translatedConfig);
+
+        if (newConfig.strategy.type === "ttl") {
+          newConfig.strategy = populateMinDurationSeconds(newConfig.strategy);
```

Comment:
Is there a reason we're passing in newConfig into `translateConfigToAPI` before it gets mutated on line 74? If so, it's not exactly clear to me why. If it's not, it looks like a potential source of bugs. I think it's best to try to write this without doing a mutation, something like:

```suggestion
        const newConfig: CacheConfig = {
          ...baseConfig,
          strategy: validatedStrategy.type === "ttl"
            ? populateMinDurationSeconds(validatedStrategy)
            : validatedStrategy,
        };

        const translatedConfig = translateConfigToAPI(newConfig);
        await CacheConfigApi.update(translatedConfig);
```

Reply:
This approach populates the minimum query duration in seconds a bit early, so that the API receives the number both in milliseconds (which it needs) and in seconds (which it doesn't need). That might be harmless, but I don't remember if the API will accept this additional input.